src/js/helpers/useToast.ts

```diff
@@ -0,0 +1,131 @@
+    showToastBtnClose(toastElement);
+  }
+ 
+  return new Toast(toastElement, { autohide: options.autohide, delay: options.delay });
+}
+
+const getToastElement = (): HTMLElement => {
+  const toastContainer = document.querySelector<HTMLElement>(selectorsMap.toast.container);
+
+  if (toastContainer) {
+    const toastTemplate = toastContainer.querySelector<HTMLTemplateElement>(selectorsMap.toast.template);
+    const toastClone = toastTemplate?.content.cloneNode(true) as DocumentFragment;
+    const toastElement = toastClone?.querySelector<HTMLElement>(selectorsMap.toast.toast);
+    const toastBody = toastElement?.querySelector<HTMLElement>(selectorsMap.toast.body);
+
+    if (toastElement && toastBody) {
+      toastContainer.appendChild(toastElement);
+  
+      return toastElement;
+    } else {
+      toastContainer.remove();
+    }
+  }
+
+  return getFallbackToastElement();
+}
+
+const getFallbackToastElement = (): HTMLElement => {
+  const body = document.querySelector<HTMLBodyElement>('body');
+  const dummyElement = document.createElement('div');
+  dummyElement.innerHTML = Toaster.Fallback;
+  const toastContainer = dummyElement.querySelector<HTMLElement>(selectorsMap.toast.container);
+  const toastTemplate = toastContainer?.querySelector<HTMLTemplateElement>(selectorsMap.toast.template);
+
+  if (body && toastContainer && toastTemplate) {
+    const toastClone = toastTemplate.content.cloneNode(true) as DocumentFragment;
+    const toastElement = toastClone.querySelector<HTMLElement>(selectorsMap.toast.toast);
+
+    if (toastElement) {
+      toastContainer.appendChild(toastElement);
+      body.appendChild(toastContainer);
+  
+      return toastElement;
+    }
+  }
+
+  throw new DOMException('The object can not be cloned.');
+}
+
+
+const addToastMessage = (toastElement: HTMLElement, message: string) => {
+  const toastBody = toastElement.querySelector<HTMLElement>(selectorsMap.toast.body);
+
+  if (toastBody) {
+    toastBody.innerHTML = message;
+  }
+}
+
+const addToastClassList = (toastElement: HTMLElement, options: Toaster.Option) => {
+  let bsClassList: string = Toaster.Theme[options.type];
+  const customClassList = options.classlist;
+  
+  if (customClassList) {
+    bsClassList = bsClassList.concat(' ', customClassList.trim());
+  }
+
+  bsClassList.split(' ').forEach((value) => {
+    if (value) {
+      toastElement.classList.add(value);
+    }
+  });
+}
+
+const showToastBtnClose = (toastElement: HTMLElement) => {
+  const btnClose = toastElement.querySelector<HTMLElement>(selectorsMap.toast.close);
+
+  if (btnClose) {
+    let toastColor = toastElement.classList.toString().match(/text-\w+/)?.toString();
+
+    if (toastColor) {
+      toastColor = toastColor.substring(toastColor.indexOf('-') + 1);
+
+      if (toastColor === 'white' || toastColor === 'light') {
+        btnClose.classList.add('btn-close-white');
+      }
+    }
+
+    btnClose.classList.remove('d-none');
+  }
+}
+
+export default useToast;
```

Comment:
Just in order to remind me later, we'll need to expose all these components system to a global `window` attribute so anyone can use it without including core files of the theme